src/mscorlib/src/System/Environment.cs

```diff
@@ -821,13 +821,6 @@ private static void SetEnvironmentVariableCore(string variable, string value)
             if (string.IsNullOrEmpty(value) || value[0] == '\0')
                 value = null;
 
-            if (AppDomain.IsAppXModel() && !AppDomain.IsAppXDesignMode())
-            {
-                // Environment variable accessors are not approved modern API.
-                // so we throw PlatformNotSupportedException.
```

Comment:
Other environment APIs are stubbed out for UWP to return empty string or null (look for IsAppXModel in this file). It does not make sense to remove the check here, without doing something about other Environment APIs for AppX.